src/pytest_memray/plugin.py

```diff
@@ -75,30 +76,38 @@ def cli_hist(data: Iterable[float], bins: int, *, log_scale: bool = True) -> str
 @dataclass
 class Result:
     metadata: Metadata
-    result_file: pathlib.Path
+    result_file: Path
 
 
 class Manager:
     def __init__(self, config: Config) -> None:
         self.results: dict[str, Result] = {}
         self.config = config
-        self.result_path = tempfile.TemporaryDirectory()
+        path: Path | None = config.getvalue("memray_bin_path")
+        self.result_path: TemporaryDirectory[str] | Path = path or TemporaryDirectory()
+        self._is_result_tmp: bool = path is None
+        self._run_id = uuid.uuid4().hex
 
     @hookimpl(hookwrapper=True)  # type: ignore[misc] # Untyped decorator
     def pytest_unconfigure(self, config: Config) -> Generator[None, None, None]:
         yield
-        self.result_path.cleanup()
+        if self._is_result_tmp:
+            assert isinstance(self.result_path, TemporaryDirectory)
+            self.result_path.cleanup()
 
     @hookimpl(hookwrapper=True)  # type: ignore[misc] # Untyped decorator
     def pytest_pyfunc_call(self, pyfuncitem: Function) -> object | None:
-        testfunction = pyfuncitem.obj
+        func = pyfuncitem.obj
 
-        @functools.wraps(testfunction)
+        @functools.wraps(func)
         def wrapper(*args: Any, **kwargs: Any) -> object | None:
-            name = f"{uuid.uuid4().hex}.bin"
-            result_file = pathlib.Path(self.result_path.name) / name
+            if self._is_result_tmp:
+                name = f"{uuid.uuid4().hex}.bin"
+            else:
+                name = f"{self._run_id}-{func.__module__}-{func.__qualname__}.bin"
```

Comment:
Maybe we could always use this version? Is this good enough to handle parametrization?

Reply:
> Maybe we could always use this version? 

We could run into long path issues with this 🤔 So considered safer to go with shorter unless there's a reason for doing the more readable name.

> Is this good enough to handle parametrization?

Hmm, probably not, should test it 🤔